app/decorators/user_decorator.rb

```diff
@@ -0,0 +1,14 @@
+class UserDecorator < Draper::Decorator
+  delegate_all
+
+  def follow_or_unfollow_button(current_user)
```

Comment:
Since this is a view specific concern, I prefer to use a helper method for this myself. You are not modifying a property on the user in preparation for output, which is usually what the decorator is  used for. I can also see arguments for this version as well though.